File: Java/src/gov/nasa/larcfm/ACCoRD/ACCoRDConfig.java
/* ====================================================================
 * Authors:      George Hagen, Rick Butler and Jeff Maddalon,
 * Organization: NASA/Langley Research Center
 * Website:      http://shemesh.larc.nasa.gov/fm/
 * 
 * Copyright (c) 2011-2017 United States Government as represented by
 * the National Aeronautics and Space Administration.  No copyright
 * is claimed in the United States under Title 17, U.S.Code. All Other
 * Rights Reserved.
 * ====================================================================
 */

package gov.nasa.larcfm.ACCoRD;
import gov.nasa.larcfm.Util.Units;
  /**
   * This class captures the default parameters for ACCORD.
   *
   */
public final class ACCoRDConfig {

    public static final boolean debugPrintTime = false; // ************* makes it easier to read runtest results 
	public static boolean NewRepulsiveCrit = false; 
	// NOTE [CAM]. When flag is set to true, the new horizontal and vertical repulsive criteria are used
	
    public static boolean allowSpeedOnlyResolutions = false;
	  	
    static public double minHorizExitSpeedLoS_default = Units.from("kn",100.0);
    static public double minVertExitSpeedLoS_default = Units.from("fpm",1499.99999999);
    static public double maxGs_default = Units.from("kn",700.0);
    static public double minGs_default = Units.from("kn",150.0);       // must be greater than 0
    static public double maxVs_default = Units.from("fpm",5000.0);
    
	static public double gsSearchLosDiscard = Units.from("nmi",1.5);
	static public double vsDiscretization_default = Units.from("fpm",10.0);
	
	final public static double NMAC_D = Units.from("ft",500); // Defined in RTCA SC-147
	final public static double NMAC_H = Units.from("ft",100); // Defined in RTCA SC-147

	public static void debugln(String s) {
	  if (debugPrintTime) {
	    System.out.println(s);
	  }
	}
	
	public static void setNewRepulsiveCrit(boolean flag) {
	  NewRepulsiveCrit = flag;
	} 

	public static boolean getNewRepulsiveCrit() {
	  return NewRepulsiveCrit;
	} 
	
    // internal units right now
    public static void setGsSearchLosDiscard(double val) {
       gsSearchLosDiscard = val;
    }

    public static void setAllowSpeedOnlyResolutions(boolean flag) {
      allowSpeedOnlyResolutions = flag;
    }

}

File: Modules/ACCoRD/src/ACCoRDConfig.cpp
/* ====================================================================
 * Authors:      George Hagen, Rick Butler and Jeff Maddalon,
 * Organization: NASA/Langley Research Center
 * Website:      http://shemesh.larc.nasa.gov/fm/
 *
 * Copyright (c) 2011-2020 United States Government as represented by
 * the National Aeronautics and Space Administration.  No copyright
 * is claimed in the United States under Title 17, U.S.Code. All Other
 * Rights Reserved.
 * ====================================================================
 *
 * ACCoRDConfig 
 *
 */

#include "ACCoRDConfig.h"
#include "Units.h"

namespace larcfm {

const double ACCoRDConfig::minHorizExitSpeedLoS_default = Units::from("kn",100.0);
const double ACCoRDConfig::minVertExitSpeedLoS_default = Units::from("fpm",1499.99999999);
const double ACCoRDConfig::maxGs_default = Units::from("kn",700.0);
const double ACCoRDConfig::minGs_default = Units::from("kn",150.0);       // must be greater than 0
const double ACCoRDConfig::maxVs_default = Units::from("fpm",5000.0);       // must be greater than 0
double ACCoRDConfig::gsSearchLosDiscard = Units::from("nm",1.5);
double ACCoRDConfig::vsDiscretization_default = Units::from("fpm",10.0);

const double ACCoRDConfig::NMAC_D = Units::from("ft",500); // Defined in RTCA SC-147
const double ACCoRDConfig::NMAC_H = Units::from("ft",100); // Defined in RTCA SC-147

// internal units right now
void ACCoRDConfig::setGsSearchLosDiscard(double val) {
  gsSearchLosDiscard = val;
}


}


